indico/modules/events/abstracts/forms.py

```diff
@@ -474,7 +474,10 @@ def __init__(self, *args, **kwargs):
         if abstracts_settings.get(self.event, 'contrib_type_required'):
             inject_validators(self, 'submitted_contrib_type', [DataRequired()])
         super(AbstractForm, self).__init__(*args, **kwargs)
-        self.submitted_contrib_type.query = self.event.contribution_types
+        if self.event.can_manage(session.user):
```

Comment:
I would use the `management` argument that's available in `make_abstract_form` for this (pass it to the form constructor). No need to check the event ACL again.